tgui/packages/tgui/interfaces/BugReportForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { BooleanLike } from 'common/react';
+import React, { createRef, useState } from 'react';
+
+import { useBackend } from '../backend';
+import { Flex, Section } from '../components';
+import { ButtonCheckbox } from '../components/Button';
+import { Window } from '../layouts';
+interface FormTypes {
+  awaiting_admin_approval: BooleanLike;
+  report_details: FormDetails;
+}
+
+// all the information necessary to pass into the github api
+type FormDetails = {
+  steps: string;
+  title: string;
+  description: string;
+  expected_behavior: string;
+};
+
+const InputTitle = (props) => {
+  return (
+    <h2>
+      {props.children}
+      {props.required && <span className="input-title-required">{' *'}</span>}
+    </h2>
+  );
+};
+
+export const BugReportForm = (props) => {
+  const { act, data } = useBackend<FormTypes>();
+  const { awaiting_admin_approval, report_details } = data;
+  const [checkBox, setCheckbox] = useState(false);
+
+  const [title, setTitle] = useState(report_details?.title || '');
+  const [steps, setSteps] = useState(report_details?.steps || '');
+  const [description, setDescription] = useState(
+    report_details?.description || '',
+  );
+  const [expected_behavior, setExpectedBehavior] = useState(
+    report_details?.expected_behavior || '',
+  );
+
+  const titleRef = createRef<HTMLInputElement>();
+  const stepsRef = createRef<HTMLTextAreaElement>();
+  const descriptionRef = createRef<HTMLInputElement>();
+  const expectedBehaviorRef = createRef<HTMLInputElement>();
```

Comment:
where are these refs used

Reply:
Nowhere, Its use was factored out earlier when I switched to useState instead; I forgot to remove it altogether.